fahrschein/src/main/java/org/zalando/fahrschein/NakadiClient.java

```diff
@@ -54,6 +58,35 @@ public static NakadiClientBuilder builder(URI baseUri) {
         }
     }
 
+    public <T> void publish(String eventName, List<T> events) throws IOException {
+        final URI uri = baseUri.resolve(String.format("/event-types/%s/events", eventName));
+        final ClientHttpRequest request = clientHttpRequestFactory.createRequest(uri, HttpMethod.POST);
+
+        request.getHeaders().setContentType(MediaType.APPLICATION_JSON);
+
+        try (final OutputStream body = request.getBody()) {
+            objectMapper.writeValue(body, events);
+        }
+
+        try (final ClientHttpResponse response = request.execute()) {
+            if ((response.getRawStatusCode() == 207 || response.getRawStatusCode() == 422)) {
```

Comment:
Yes, that's the big downside of handling status codes problems inside that wrapper, it already caused me some hard to debug issues. The alternative would be to do these status checks in every method doing a request, maybe using a static helper. I'll think about that.